src/libsystemd-network/sd-ipv4acd.c

```diff
@@ -73,8 +74,10 @@ struct sd_ipv4acd {
         void* userdata;
 };
 
-#define log_ipv4acd_errno(acd, error, fmt, ...) log_internal(LOG_DEBUG, error, PROJECT_FILE, __LINE__, __func__, "IPV4ACD: " fmt, ##__VA_ARGS__)
-#define log_ipv4acd(acd, fmt, ...) log_ipv4acd_errno(acd, 0, fmt, ##__VA_ARGS__)
+#define log_ipv4acd_errno(acd, error, fmt, ...)                         \
+        log_interface_full_errno(acd ? sd_ipv4acd_get_ifname(acd) : NULL, LOG_DEBUG, error, "IPV4ACD: " fmt, ##__VA_ARGS__)
```

Comment:
Unfortunately this does double evaluation of `acd`. So maybe make `sd_ipv4acd_get_ifname()` just return NULL if the argument is NULL without complaining?